metals/src/main/scala/scala/meta/internal/metals/FileWatcher.scala

```diff
@@ -67,11 +67,13 @@ final class FileWatcher(
         } else {
           path
         }
-        pathToCreate.createDirectories()
+        val createdPaths = pathToCreate.createDirectories()
```

Comment:
you prefer to have a separate method that takes a path and getAllAbsolutePathToCreate ? 
How can you inline without verifying the existence of a file ?

We can add tests to make sure that the old an new version of createDirectories have the same behavior ?

Reply:
I would prefer a separate method for sure. I believe it could be added here, I don't think it's something we will use besides this case.

No need to add additional tests. I just think it will be more optimized to create all dirs at the same time by default. Usually it's a single filesystem operation.

Reply:
It's not. The implementation creates all nonexistent parent directories first.

Reply:
That's right, that's why I think it's better to use the new more complicated method only in case of the FileWatcher.

Reply:
Ok, I actually misunderstood. You meant that Java implementation is doing the same as the new here. We can leave it then.